src/app/view/components/dot-navigation/dot-navigation.service.ts

```diff
@@ -18,11 +19,16 @@ export class DotNavigationService {
         private dotRouterService: DotRouterService,
         private dotcmsEventsService: DotcmsEventsService,
         private loginService: LoginService,
-        private location: PlatformLocation
+        private location: PlatformLocation,
+        private router: Router
     ) {
-        this.dotMenuService.loadMenu().subscribe((menu: DotMenu[]) => {
-            this.setMenu(menu);
-        });
+        router.events
```

Comment:
Load menu when Route Navigation ends if is a private page

Reply:
I'm not sure about this but, doesn't this will trigger every time we navigate to one place to another? and also, why do we need this ```NavigationEnd``` anyways?

Reply:
The subscription to the NavigationEnd, is needed, because we need to update the menu item once the route is served and when non public page. In the previous scenario we were asking for a menu update before the route was updated, and that was causing issues with the selected option.